components/net_traits/request.rs

```diff
@@ -198,6 +199,101 @@ impl Default for RequestInit {
             url_list: vec![],
         }
     }
+
+    pub fn method(mut self, method: Method) -> RequestInit {
```

Comment:
This looks like it is a consuming builder instead?

Reply:
Yeah, sorry, I switched the names. xD

Well, Actually I didn't implemented a builder pattern, since it hasn't a Builder class. I've created a constructor, called new, and setters methods, that return the object. Using this approach, I don't need this Builder class, it will have more readability, since it hasn't an extra class. I'm not sure about its performance.

What you think? Can I use this approach? or you really want a builder?

Reply:
What you just described is the Rust builder pattern. Classes do not exist in Rust, and I've often understood that `RequestInit` is already a builder struct for `Request`. Perhaps renaming it `RequestBuilder` would make it more clear and apparent?

Reply:
Sure

Reply:
It confused me a lot, whose name, I was thinking that you were looking for a builder to the TYPE RequestInit, I didn't figure out, that the builder was the RequestInit.
I gonna make the changes as soon as possible.

And how about the builder, do you want a  non-consuming builder or a consuming builder?

Reply:
I don't have any opinions either way, but I think the guide you referenced has a preference to non-consuming builders?